third_party/netcdf-c.BUILD

```diff
@@ -12,30 +12,983 @@
+        "libdispatch/nchashmap.c",
+        "libdispatch/ncjson.c",
+        "libdispatch/nclist.c",
+        "libdispatch/nclistmgr.c",
+        "libdispatch/nclog.c",
+        "libdispatch/nctime.c",
+        "libdispatch/ncuri.c",
+        "libdispatch/ncxcache.c",
+        "libdispatch/utf8proc.c",
+        "libdispatch/utf8proc.h",
+    ],
+    hdrs = hdrs,
+    defines = defines,
+    includes = [
+        "include",
+        "libdispatch",
+    ],
+    textual_hdrs = [
+        # breaks build because of undefined type
+        "libdispatch/utf8proc_data.c",
+    ],
+    deps = ["@hdf5"],
+)
+
+genrule(
+    name = "gen_attr_c",
+    srcs = ["libsrc/attr.m4"],
+    outs = ["attr.c"],
+    cmd = "m4 $(SRCS) > $@",
+)
+
+genrule(
+    name = "gen_ncx_c",
+    srcs = ["libsrc/ncx.m4"],
+    outs = ["ncx.c"],
+    cmd = "m4 $(SRCS) > $@",
+)
+
+genrule(
+    name = "gen_putget_c",
+    srcs = ["libsrc/putget.m4"],
+    outs = ["putget.c"],
+    cmd = "m4 $(SRCS) > $@",
+)
+
+cc_library(
+    name = "netcdf3",
+    srcs = [
+        "libsrc/dim.c",
+        "libsrc/lookup3.c",
+        "libsrc/memio.c",
+        "libsrc/mmapio.c",
+        "libsrc/nc3dispatch.c",
+        "libsrc/nc3internal.c",
+        "libsrc/ncio.c",
+        "libsrc/ncio.h",
+        "libsrc/ncx.h",
+        "libsrc/posixio.c",
+        "libsrc/pstdint.h",
+        "libsrc/v1hpg.c",
+        "libsrc/var.c",
+        ":gen_attr_c",
+        ":gen_ncx_c",
+        ":gen_putget_c",
     ],
+    hdrs = hdrs,
+    defines = defines,
+    # Allows including <config.h> with angle brackets
+    include_prefix = ".",
+    includes = [
+        "include",
+        "libsrc",
+    ],
+)
+
+cc_library(
+    name = "netcdf4",
+    srcs = [
+        "libsrc4/nc4attr.c",
+        "libsrc4/nc4cache.c",
+        "libsrc4/nc4dim.c",
+        "libsrc4/nc4dispatch.c",
+        "libsrc4/nc4grp.c",
+        "libsrc4/nc4internal.c",
+        "libsrc4/nc4type.c",
+        "libsrc4/nc4var.c",
+        "libsrc4/ncfunc.c",
+        "libsrc4/ncindex.c",
+    ],
+    hdrs = hdrs,
+    defines = defines,
+    includes = ["include"],
+    deps = ["@hdf5"],
+)
+
+cc_library(
+    name = "netcdfhdf5",
+    srcs = [
+        "libhdf5/hdf5attr.c",
+        "libhdf5/hdf5create.c",
+        "libhdf5/hdf5debug.c",
+        "libhdf5/hdf5debug.h",
+        "libhdf5/hdf5dim.c",
+        "libhdf5/hdf5dispatch.c",
+        "libhdf5/hdf5err.h",
+        "libhdf5/hdf5file.c",
+        "libhdf5/hdf5filter.c",
+        "libhdf5/hdf5grp.c",
+        "libhdf5/hdf5internal.c",
+        "libhdf5/hdf5open.c",
+        "libhdf5/hdf5set_format_compatibility.c",
+        "libhdf5/hdf5type.c",
+        "libhdf5/hdf5var.c",
+        "libhdf5/nc4hdf.c",
+        "libhdf5/nc4info.c",
+        "libhdf5/nc4mem.c",
+        "libhdf5/nc4memcb.c",
+    ],
+    hdrs = hdrs,
+    defines = defines,
+    includes = ["include"],
+    deps = ["@hdf5"],
+)
+
+genrule(
+    name = "gen_config_h",
+    outs = ["config.h"],
+    cmd = r"""
```

Comment:
I'd like to move this to a variable so that we can `select` the string that corresponds to the platform we're building.

We'll probably need an AND chain to express the logic: https://bazel.build/docs/configurable-attributes#and-chaining